src/foam/u2/detail/SectionedDetailPropertyView.js

```diff
@@ -355,5 +362,13 @@ foam.CLASS({
             .end();
         }));
     }
+  ],
+
+  listeners: [
+    function loaded() {
+      // Mark that we have descendant elements of the same class. If this is
+      // true, we don't want to add any padding to this element.
+      this.hasDescendants = this.el().querySelector('.foam-u2-detail-SectionedDetailPropertyView') !== null;
```

Comment:
What if we remove the padding around this Model completely and make it the SectionedDetailView's responsibility to put padding between the children. Don't put top padding on the first child or bottom padding on the last. This may be annoying to fix up everywhere but it feels more correct to me. What do you think?

Reply:
I can look into it. One thing to note though is that we have to deal with `VerticalDetailView` being nested as well, but we can just do the same thing.

Reply:
Ah yeah good catch. It wouldn't surprise me for others to be affected too and even if you don't catch them all here we can fix them as we see them.

Reply:
I've been trying this out and there are two tricky parts so far:

1. Selecting the last `.foam-u2-detail-SectionedDetailPropertyView` in a `.foam-u2-detail-SectionView` isn't possible with CSS since `:last-of-type` doesn't apply to CSS classes, just simple element selectors like `div` or `p`. So since we have other `div`s after the `.foam-u2-detail-SectionedDetailPropertyView`, that won't work. The recommended solution I saw on StackOverflow was to use JavaScript to mark the last one with an extra CSS class. Another idea is to switch from `div`s to `section`s for `.foam-u2-detail-SectionedDetailPropertyView`s. Even in that case, point 2 below is still an issue.
2. Knowing which one is the last one isn't as simple as I thought since each property is shown/hidden based on a visibility slot. So the last property in the array isn't necessarily the last _visible_ property in the array. We can figure this out with some work, but at that point I don't know if it's a better or worse solution than the one I have open in this PR currently.

Reply:
After talking with @mcarcaso in person, we want to try this:

* `SectionView` will create the visibility slots for each of its child `SectionedDetailPropertyView`s and pass them via a new `visibilitySlot` property on `SectionDetailedPropertyView`
  * We don't want to break anything else that depends on `SectionDetailedPropertyView` taking care of its own visibility, so if a `SectionDetailedPropertyView`'s `visibilitySlot` property is set then it'll use that, otherwise it'll make its own (like it's already doing). We could also just duplicate it and it would technically be fine, but it's more efficient just to create one slot instead of two duplicate ones.
  * Since `SectionView` has access to all of its children's visibility slots, it can figure out which of its children are the first and last visible ones and style them accordingly (no top-padding on the first visible child and no bottom-border on the last visible chid)

Reply:
Implemented in 7dab75a.